### PR TITLE
Attempt to speed up tests issue #229

### PR DESCRIPTION
**Pull request by liosedhel**
First approach, split test using strongly connected component parameter (computed for each node).
Not sure if this is the best approach though, as what we really care is leveraging testing time (not grouping dependencies).
zinc_dep_graph

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
Hi @liosedhel,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by jvican** [comment]
@liosedhel Great, great job.
  
I particularly like that graph, it's obvious you've poured some hours into this! I've added some commits to your branch to fix the issues of the CI and I've also taken the liberty to cache Travis sbt dirs, so that CI is even faster.

File: .travis.yml [hunk 1]
```diff
@@ -4,8 +4,19 @@ scala:
   - 2.11.8
   - 2.12.1
 
+cache:
+  directories:
+    - $HOME/.ivy2/cache
+    - $HOME/.sbt/boot
```
Comment (dwijnand) [inline .travis.yml]: @jvican It's better (and I believe safe) to cache the whole ~/.sbt directory. That way you cache the launcher in ~/.sbt/launchers

Comment (jvican) [reply .travis.yml]: Cool, I got the commands from https://github.com/typesafehub/zinc/blob/master/.travis.yml. Happy to know we also cache launcher the way you propose.

Comment (dwijnand) [reply .travis.yml]: I/someone should update http://www.scala-sbt.org/0.13/docs/Travis-CI-with-sbt.html#%28Experimental%29+Reusing+Ivy+cache

Also could you include the `before_cache` part too please, for correctness:

```yml
before_cache:
  - find $HOME/.ivy2/cache -name "ivydata-*.properties" -print -delete
  - find $HOME/.sbt        -name "*.lock"               -print -delete
```

Comment (jvican) [reply .travis.yml]: Sure, I have no idea how Travis works. These tips are nice.

**Comment by jvican** [comment]
@liosedhel You've basically decreased our build times from 40 long minutes to 6, that is a ~7x speedup. Thanks for taking the time to cluster correctly the tasks. This improvement is awesome.

File: .travis.yml (hunk 1 above)
Comment (dwijnand) [reply .travis.yml]: https://github.com/sbt/website/pull/328

**Comment by liosedhel** [comment]
@jvican Thank you for your help during ScalaSphere hackaton as also for finishing this PR. I've done only a small part of the job, can't take a credit ;) Hope that this PR will not broke anything ;)

**Review by dwijnand — APPROVED** [review]
My only concern with this is that this shifts the responsibility of what is being run more towards the Travis CI config (e.g not simply delegating to the aggregation in the sbt build).

But who can argue with those amazing numbers?!

Let's just remember from now on if/when changing the build layout to also change .travis.yml.

Thank you @liosedhel and welcome to Zinc!

**Comment by dwijnand** [comment]
Seeking another maintainer's :+1: to merge this.

**Review by romanowski — APPROVED** [review]
LGTM

I will probably use this approach in my plugin.

**Comment by romanowski** [comment]
@liosedhel @dwijnand How was graph generated?

**Comment by liosedhel** [comment]
@romanowski I've just manually written down all the dependencies between modules, using CSV format, and then imported it to the Gephi. Then tried different layouts and run some algorithms (to get strongly connected component parameter), modified a view (to add size and colors) and voilà :) Can send you a gephi project.

**Comment by dwijnand** [comment]
There's also https://github.com/dwijnand/sbt-project-graph

**Pull request merged**